main/src/main/java/cgeo/geocaching/CacheDetailActivity.java

```diff
@@ -1847,6 +1848,46 @@ public void setContent() {
 
         }
 
+        private void translateListing() {
+            final CacheDetailActivity cda = (CacheDetailActivity) getActivity();
+
+            if (cda.translationStatus.isTranslated()) {
```

Comment:
I don't understand this if statement.

If I understand the code of "Status" correctly, then translateListing() is supposed to be called (via callback) whenever a new translation is needed (eg because source or target language changed or necessary download is done etc etc) and can be performed.
* Why is then logic here to check and maybe correct translation statusÜ
* Why is the method left when translation is done but at the same time "translation not done" marked?
* What is the meaning of the strange logic for "hint" here?

Reply:
This is the "if already translated restore the original, untranslated listing".